packages/block-library/src/gallery/gallery-image.js

```diff
@@ -281,6 +292,7 @@ class GalleryImage extends Component {
 						}
 						unstableOnFocus={ this.onSelectCaption }
 						inlineToolbar
+						id={ 'caption' + id }
```

Comment:
There's a typo here:
```suggestion
						id={ 'caption-' + id }
```